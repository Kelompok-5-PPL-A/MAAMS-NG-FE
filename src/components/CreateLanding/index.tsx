import { CustomInput } from '../customInput'
import Link from 'next/link'
import React, { useEffect, useState } from 'react'
import { useRouter } from 'next/router'
import toast from 'react-hot-toast'

const CreateLanding = () => {
  const [isLoggedIn, setIsLoggedIn] = useState(false)
  const [question, setQuestion] = useState<string>('')
  const router = useRouter()

  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault()

    if (!question) {
      toast.error('Pertanyaan harus diisi')
      return
    }

    router.push({
      pathname: '/validator',
      query: { question: question }
    })
  }

  useEffect(() => {
    const refresh_token = localStorage.getItem('isLoggedIn')
    setIsLoggedIn(refresh_token === 'true')
  })

  return (
    <div className='flex flex-col lg:flex-row justify-between items-center gap-12 lg:mx-55'>
      <div className=''>
        <img src='/icons/landing-icon.svg' alt='landing' className='' />
      </div>
      <div className=''>
          <div className='flex flex-col gap-8 mx-12'>
            <h1 className='text-3xl font-bold text-center'>Apa masalah yang ingin dianalisis hari ini?</h1>
            <div className='flex flex-row gap-2'>
              <form className='flex flex-row w-full gap-x-4 items-center justify-center' onSubmit={handleSubmit}>
                <CustomInput
                  inputClassName='p-6 text-sm w-[400px] border border-black-300 rounded-lg shadow-md focus:shadow-lg transition-shadow duration-200'
                  placeholder='ingin menganalisis apa hari ini ...'
                  spacerClassName='space-y-0'
                  isDisabled={false}
                  onChange={(e) => setQuestion(e.target.value)}
                  value={question}
                />
                <button className='' title='submit_button'>
                  <img src='/icons/send-icon.svg' alt='search' className='bg-yellow-400 p-3 w-20 h-15 rounded-full' />
                </button>
              </form>
            </div>
          </div>
      </div>
    </div>
  )
}

export default CreateLanding