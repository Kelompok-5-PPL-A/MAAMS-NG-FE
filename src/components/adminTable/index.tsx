import React from 'react'
import { Item } from '../../components/types/adminTable'
import { useRouter } from 'next/router'

interface Props {
  data: Item[]
}

const AdminTable: React.FC<Props> = ({ data }) => {
  const router = useRouter()

  const handleRowClick = (id: string) => {
    router.push(`/validator/${id}`)
  }

  return (
    <div className='relative overflow-x-auto shadow-md m-12'>
      <table className='w-full text-sm text-left text-black border-collapse'>
        <thead className='text-xs uppercase bg-[#FBC707] text-black'>
          <tr>
            <th scope='col' className='px-6 py-3 w-1/5'>
              Judul
            </th>
            <th scope='col' className='px-6 py-3 w-1/5'>
              Pengguna
            </th>
            <th scope='col' className='px-6 py-3 w-1/5'>
              Topik
            </th>
            <th scope='col' className='px-6 py-3 w-1/5'>
              Waktu
            </th>
            <th scope='col' className='px-6 py-3 w-1/5'>
              <span className='sr-only'>Lihat</span>
            </th>
          </tr>
        </thead>
        <tbody>
          {data.map(({ id, title, displayed_title, user, tags, timestamp }) => (
            <tr
              key={id}
              className='bg-white hover:bg-gray-200 border border-[#FBC707] text-xs h-20 cursor-pointer'
              onClick={() => handleRowClick(id)}
            >
              <td className='px-6 py-4 border-b border-b-[#FBC707] truncate'>
                {(displayed_title || title).length > 60
                  ? `${(displayed_title || title).slice(0, 60)}...`
                  : displayed_title || title}
              </td>
              <td className='px-6 py-4 border-b border-b-[#FBC707] truncate'>
                {user?.length > 10 ? `${user.slice(0, 10)}...` : user}
              </td>
              <td className='px-6 py-4 flex justify-start'>
                {tags.map((tag, tagIndex) => (
                  <button
                    key={tagIndex}
                    className='bg-[#FBC707] text-black px-3 py-3 rounded-3xl text-xs font-bold mr-3'
                  >
                    {tag}
                  </button>
                ))}
              </td>
              <td className='px-6 py-4 border-b border-b-[#FBC707]'>{timestamp}</td>
              <td className='px-6 py-4 text-right border-b border-b-[#FBC707]'>
                <span
                  className='text-blue-600 underline cursor-pointer hover:text-sm'
                  data-testid={`view-button-${id}`}
                  onClick={(e) => {
                    e.stopPropagation()
                    handleRowClick(id)
                  }}
                >
                  Lihat
                </span>
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  )
}

export default AdminTable
