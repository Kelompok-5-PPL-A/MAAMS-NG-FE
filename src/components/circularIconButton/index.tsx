import React from 'react'
import { CircularIconButtonProps } from '../types/circularIconButtonProps'

export const CircularIconButton: React.FC<CircularIconButtonProps> = ({ id, icon, onClick, type }) => {
  return (
    <div className='w-[52px] h-[52px] bg-yellow-400 rounded-full flex justify-center items-center'>
      <button data-testid='submit-question' id={id} type={type} color='black' onClick={onClick}>
        {icon}
      </button>
    </div>
  )
}
