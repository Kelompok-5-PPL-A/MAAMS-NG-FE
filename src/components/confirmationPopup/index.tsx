import React, { useState, useEffect } from 'react'
import { ConfirmationPopupProps } from '../types/confirmation'
import Mode from '../../constants/mode'

const Dialog: React.FC<{ isOpen: boolean; onClose: () => void; children: React.ReactNode }> = ({ isOpen, onClose, children }) => {
    useEffect(() => {
      const handleKeyDown = (e: KeyboardEvent) => {
        if (e.key === 'Escape') {
          onClose()
        }
      }
      if (isOpen) {
        document.addEventListener('keydown', handleKeyDown)
      }
      return () => {
        document.removeEventListener('keydown', handleKeyDown)
      }
    }, [isOpen, onClose])
  
    if (!isOpen) return null
  
    return (
      <div
        className='fixed inset-0 bg-black bg-opacity-50 flex items-center justify-center z-50' 
        onClick={onClose} 
        role='dialog' 
        aria-modal='true' 
        tabIndex={0}
        onKeyDown={(e) => e.key === 'Enter' && onClose()}
      >
        <div
          className='bg-white w-[400px] max-w-full p-6 rounded-lg shadow-lg overflow-auto text-center' 
          onClick={(e) => e.stopPropagation()} 
          role='document'
          tabIndex={0}
        >
          {children}
        </div>
      </div>
    )
  }

const Button: React.FC<{ onClick: () => void; variant?: 'outline' | 'gradient'; children: React.ReactNode }> = ({
  onClick,
  variant = 'outline',
  children,
}) => {
  const baseClass = 'py-2 px-12 rounded-xl font-semibold'
  const variantClass = variant === 'gradient' ? 'bg-gradient-to-b from-yellow-400 to-yellow-600 text-l text-white font-bold py-2 px-12 rounded-xl' : 'border border-yellow-400 text-yellow-400'

  return (
    <button onClick={onClick} className={`${baseClass} ${variantClass}`}>
      {children}
    </button>
  )
}

const ConfirmationPopup: React.FC<ConfirmationPopupProps> = ({ mode, onConfirm, onCancel }) => {
  const [isOpen, setIsOpen] = useState(true)

  const handleConfirm = () => {
    onConfirm()
    setIsOpen(false)
  }

  const handleCancel = () => {
    onCancel()
    setIsOpen(false)
  }

  const getMessage = () => {
    return mode === Mode.pengawasan
      ? 'Apakah Anda yakin ingin menampilkan analisis ini kepada Admin?'
      : 'Ubah analisis menjadi pribadi?'
  }

  return (
    <Dialog isOpen={isOpen} onClose={handleCancel}>
      <h2 className='text-xl font-bold mb-4'>{getMessage()}</h2>
      <div className='flex justify-center gap-4'>
        <Button variant='outline' onClick={handleCancel}>
          Batal
        </Button>
        <Button variant='gradient' onClick={handleConfirm}>
          Simpan
        </Button>
      </div>
    </Dialog>
  )
}

export default ConfirmationPopup
