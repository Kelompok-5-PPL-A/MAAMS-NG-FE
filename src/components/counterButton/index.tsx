import { CounterButtonProps } from '../types/counterButton'
import React from 'react'

export const CounterButton: React.FC<CounterButtonProps> = ({ number, onIncrement, onDecrement }) => {
  const textStyle = 'text-2xl font-bold relative text-center whitespace-nowrap z-[1]'
  const decrementBorderBase =
    'border-r-0 flex h-10 py-2 px-8 justify-center items-center shrink-0 flex-nowrap rounded-tl-[10px] rounded-tr-none rounded-br-none rounded-bl-[10px] border-solid border relative'
  const incrementBorderBase =
    'border-l-0 flex h-10 py-2 px-8 justify-center items-center shrink-0 flex-nowrap rounded-tl-none rounded-tr-[10px] rounded-br-[10px] rounded-bl-none border-solid border relative z-[4]'

  let decrementStyle = 'text-[#fbc707] ' + textStyle
  let incrementStyle = 'text-[#fbc707] ' + textStyle
  let decrementBorderStyle = 'border-[#fbc707] ' + decrementBorderBase
  let incrementBorderStyle = 'border-[#fbc707] ' + incrementBorderBase

  if (number === 3) {
    decrementStyle = 'text-[#aeaeae] ' + textStyle
    decrementBorderStyle = 'border-[#aeaeae] ' + decrementBorderBase
  } else if (number === 5) {
    incrementStyle = 'text-[#aeaeae] ' + textStyle
    incrementBorderStyle = 'border-[#aeaeae] ' + incrementBorderBase
  }

  return (
    <div className='items-center my-2'>
      <h2 className='text-center text-sm font-bold text-black relative mx-auto mb-2'>Jumlah Kolom (Max 5)</h2>
      <div className='main-container flex justify-center items-center flex-nowrap relative mx-auto my-0'>
        <button onClick={onDecrement} className={decrementBorderStyle}>
          <span className={decrementStyle}>-</span>
        </button>
        <div className='flex h-10 py-2 px-8 justify-center items-center shrink-0 flex-nowrap bg-[#fbc707] relative z-[2]'>
          {number >= 0 && (
            <span className='shrink-0 basis-auto text-[14px] font-bold text-black relative text-left whitespace-nowrap z-[3]'>
              {number}
            </span>
          )}
        </div>
        <button onClick={onIncrement} className={incrementBorderStyle}>
          <span className={incrementStyle}>+</span>
        </button>
      </div>
    </div>
  )
}
