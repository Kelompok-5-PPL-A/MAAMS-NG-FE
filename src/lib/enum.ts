export enum CauseStatus {
    Incorrect = 'INCORRECT',
    CorrectNotRoot = 'CORRECT_NOT_ROOT',
    CorrectRoot = 'CORRECT_ROOT',
    Resolved = 'RESOLVED',
    Unchecked = 'UNCHECKED'
  }
  