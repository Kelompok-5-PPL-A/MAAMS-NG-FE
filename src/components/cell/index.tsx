import React from 'react'
import { CauseStatus } from '../../lib/enum'
import { CellProps } from '../types/cell'

export const Cell: React.FC<CellProps> = ({
  cellName,
  cause,
  onChange,
  causeStatus,
  disabled,
  placeholder,
  feedback
}) => {
  const getOutlineClass = (status: CauseStatus) => {
    switch (status) {
      case CauseStatus.Incorrect:
        return 'border-red-500'
      case CauseStatus.CorrectNotRoot:
        return 'border-green-500'
      case CauseStatus.CorrectRoot:
        return 'border-purple-500'
      case CauseStatus.Resolved:
        return 'border-gray-200'
      case CauseStatus.Unchecked:
      default:
        return 'border-black'
    }
  }

  const outlineClass = getOutlineClass(causeStatus)

  const renderFeedback = () => {
    let emoji = ''
    let color = ''
    let feedbackText = feedback || ''
    
    switch (causeStatus) {
      case CauseStatus.CorrectRoot:
        emoji = '☑️'
        color = 'purple'
        if (!feedbackText.includes('Akar Masalah')) {
          feedbackText = `Akar Masalah Kolom ${cellName[0]} ditemukan: ${feedbackText}`
        }
        break
      case CauseStatus.CorrectNotRoot:
        emoji = '✅'
        color = 'green'
        break
      case CauseStatus.Incorrect:
        emoji = '❌'
        color = 'red'
        break
      default:
        return null
    }

    return (
      <div className='feedback-text mt-4' style={{ color: color }}>
        {emoji} {feedbackText}
      </div>
    )
  }

  return (
    <div className='flex flex-col items-center justify-center relative'>
      <div className='relative w-fit mt-[-1.00px] font-bold text-black text-2xl leading-10 mb-2 whitespace-nowrap'>
        {cellName}
      </div>
      <textarea
        value={cause}
        onChange={(event) => {
          !disabled && onChange(event.target.value)
        }}
        rows={1}
        maxLength={148}
        className={`w-full h-22 text-xs resize-none flex pt-4 px-4 pb-16 items-center bg-[#ececec] border-solid border-2 ${outlineClass} relative z-[1]`}
        placeholder={placeholder}
        disabled={disabled}
      ></textarea>
      {renderFeedback()}
    </div>
  )
}