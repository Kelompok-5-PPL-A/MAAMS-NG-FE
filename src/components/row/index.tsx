import React, { useState, useEffect } from 'react'
import { Cell } from '../cell/index'
import { CauseStatus } from '../../lib/enum'
import { RowProps } from '../types/row'

export const Row: React.FC<RowProps> = ({
  rowNumber,
  cols,
  causes,
  causeStatuses,
  disabledCells,
  onCauseAndStatusChanges,
  feedbacks
}) => {
  const alphabet = 'ABCDE'
  const [localCauses, setLocalCauses] = useState<string[]>(causes)
  const [localCauseStatuses, setLocalCauseStatuses] = useState<CauseStatus[]>(causeStatuses)

  useEffect(() => {
    setLocalCauses(causes.slice(0, cols).concat(Array(Math.max(cols - causes.length, 0)).fill('')))
    setLocalCauseStatuses(
      causeStatuses.slice(0, cols).concat(Array(Math.max(cols - causeStatuses.length, 0)).fill(CauseStatus.Unchecked))
    )
  }, [cols, causes, causeStatuses])

  const handleLocalCauseChange = (causeIndex: number, newValue: string, newStatus: CauseStatus) => {
    const updatedCauses = [...localCauses]
    updatedCauses[causeIndex] = newValue
    setLocalCauses(updatedCauses)

    const updatedStatuses = [...localCauseStatuses]
    updatedStatuses[causeIndex] = newStatus
    setLocalCauseStatuses(updatedStatuses)

    onCauseAndStatusChanges(causeIndex, newValue, newStatus)
  }

  let gridClass = ''
  if (cols === 3) {
    gridClass = 'grid grid-cols-3 gap-0 items-stretch my-8'
  } else if (cols === 4) {
    gridClass = 'grid grid-cols-4 gap-0 items-stretch my-8'
  } else {
    gridClass = 'grid grid-cols-5 gap-0 items-stretch my-8'
  }

  return (
    <div>
      {cols >= 3 && cols <= 5 && (
        <div className={gridClass} data-testid='row-container'>
          {causes.map((cause, index) => (
            <div data-testid='cell' key={`${alphabet[index]}${rowNumber}`}>
              <Cell
                cellName={`${alphabet[index]}${rowNumber}`}
                cause={cause}
                onChange={(newValue) => handleLocalCauseChange(index, newValue, localCauseStatuses[index])}
                causeStatus={causeStatuses[index]}
                disabled={disabledCells[index]}
                placeholder={disabledCells[index] ? '' : 'Isi sebab..'}
                feedback={feedbacks[index]}
              />
            </div>
          ))}
        </div>
      )}
    </div>
  )
}
