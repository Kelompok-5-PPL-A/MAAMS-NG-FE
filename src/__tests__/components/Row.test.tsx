import '@testing-library/jest-dom'
import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { Row } from '../../components/row'
import { CauseStatus } from '../../lib/enum'

describe('Row Component', () => {
  const rowNumber = 1
  const cols = 3
  const causes = ['', '', '']
  const causeStatuses = [CauseStatus.Unchecked, CauseStatus.Unchecked, CauseStatus.Unchecked]
  const disabledCells = [false, false, false]
  const feedbacks = ['Feedback 1', 'Feedback 2', 'Feedback 3']
  const mockOnCauseAndStatusChanges = jest.fn()

  test('renders correctly with the given number of columns, causes, and statuses', () => {
    const { getByTestId, getAllByTestId } = render(
      <Row
        rowNumber={rowNumber}
        cols={cols}
        causes={causes}
        causeStatuses={causeStatuses}
        disabledCells={disabledCells}
        onCauseAndStatusChanges={mockOnCauseAndStatusChanges}
        feedbacks={feedbacks} causesId={[]}      />
    )

    expect(getByTestId('row-container')).toHaveClass(`grid grid-cols-${cols}`)
    const cellElements = getAllByTestId('cell')
    expect(cellElements).toHaveLength(cols)
  })

  test('renders with disabled cells based on disabledCells prop', () => {
    const disabledCellsTest = [true, false, true]
    const { getAllByTestId } = render(
      <Row
        rowNumber={rowNumber}
        cols={cols}
        causes={causes}
        causeStatuses={causeStatuses}
        disabledCells={disabledCellsTest}
        onCauseAndStatusChanges={mockOnCauseAndStatusChanges}
        feedbacks={feedbacks} causesId={[]}      />
    )

    const cellElements = getAllByTestId('cell')
    expect(cellElements[0].querySelector('textarea')).toBeDisabled()
    expect(cellElements[1].querySelector('textarea')).not.toBeDisabled()
    expect(cellElements[2].querySelector('textarea')).toBeDisabled()
  })

  test('does not render with an invalid number of columns', () => {
    const invalidCols = 6

    const { queryByTestId } = render(
      <Row
        rowNumber={rowNumber}
        cols={invalidCols}
        causes={Array(invalidCols).fill('')}
        causeStatuses={Array(invalidCols).fill(CauseStatus.Unchecked)}
        disabledCells={Array(invalidCols).fill(false)}
        onCauseAndStatusChanges={mockOnCauseAndStatusChanges}
        feedbacks={Array(invalidCols).fill('')} causesId={[]}      />
    )

    expect(queryByTestId('row-container')).toBeNull()
  })

  test('correctly updates causes and statuses', () => {
    const initialCauses = ['Cause 1', 'Cause 2', 'Cause 3']
    const initialStatuses = [CauseStatus.CorrectNotRoot, CauseStatus.Incorrect, CauseStatus.Unchecked]

    const { getAllByTestId } = render(
      <Row
        rowNumber={rowNumber}
        cols={cols}
        causes={initialCauses}
        causeStatuses={initialStatuses}
        disabledCells={disabledCells}
        onCauseAndStatusChanges={mockOnCauseAndStatusChanges}
        feedbacks={feedbacks} causesId={[]}      />
    )

    const cellElements = getAllByTestId('cell')
    cellElements.forEach((cell, index) => {
      const textarea = cell.querySelector('textarea')
      expect(textarea).toHaveValue(initialCauses[index])
    })
  })

  test('matches snapshot', () => {
    const { asFragment } = render(
      <Row
        rowNumber={rowNumber}
        cols={cols}
        causes={causes}
        causeStatuses={causeStatuses}
        disabledCells={disabledCells}
        onCauseAndStatusChanges={mockOnCauseAndStatusChanges}
        feedbacks={feedbacks} causesId={[]}      />
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('renders feedback text correctly', () => {
    const { getAllByTestId } = render(
      <Row
        rowNumber={rowNumber}
        cols={cols}
        causes={causes}
        causeStatuses={causeStatuses}
        disabledCells={disabledCells}
        onCauseAndStatusChanges={mockOnCauseAndStatusChanges}
        feedbacks={feedbacks}
        causesId={[]}      
      />
    )
  
    const feedbackElements = getAllByTestId('cell')
    feedbackElements.forEach((cell, index) => {
      expect(cell).toHaveTextContent(feedbacks[index])
    })
  })
  
  test('does not render feedback when empty', () => {
    const emptyFeedbacks = ['', '', '']
    
    const { queryAllByTestId } = render(
      <Row
        rowNumber={rowNumber}
        cols={cols}
        causes={causes}
        causeStatuses={causeStatuses}
        disabledCells={disabledCells}
        onCauseAndStatusChanges={mockOnCauseAndStatusChanges}
        feedbacks={emptyFeedbacks}
        causesId={[]}      
      />
    )
  
    const feedbackElements = queryAllByTestId('cell')
    feedbackElements.forEach((cell) => {
      expect(cell).not.toHaveTextContent('Feedback')
    })
  })
  
  test('calls onCauseAndStatusChanges when a cause is updated', () => {
    const { getAllByTestId } = render(
      <Row
        rowNumber={rowNumber}
        cols={cols}
        causes={causes}
        causeStatuses={causeStatuses}
        disabledCells={disabledCells}
        onCauseAndStatusChanges={mockOnCauseAndStatusChanges}
        feedbacks={feedbacks}
        causesId={[]}      
      />
    )
  
    const textareas = getAllByTestId('cell').map(cell => cell.querySelector('textarea'))
  
    fireEvent.change(textareas[1]!, { target: { value: 'Updated Cause' } })
  
    expect(mockOnCauseAndStatusChanges).toHaveBeenCalledWith(1, 'Updated Cause', CauseStatus.Unchecked)
  })
  
})