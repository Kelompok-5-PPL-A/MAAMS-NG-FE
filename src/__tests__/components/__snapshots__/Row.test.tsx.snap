// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="grid grid-cols-3 gap-0 items-stretch my-8"
      data-testid="row-container"
    >
      <div
        data-testid="cell"
      >
        <div
          class="flex flex-col items-center justify-center relative"
        >
          <div
            class="relative w-fit mt-[-1.00px] font-bold text-black text-2xl leading-10 mb-2 whitespace-nowrap"
          >
            A1
          </div>
          <textarea
            class="w-full h-22 text-xs resize-none flex pt-4 px-4 pb-16 items-center bg-[#ececec] border-solid border border-black relative z-[1]"
            maxlength="148"
            placeholder="Isi sebab.."
            rows="1"
          />
          <div
            class="feedback-text mt-4"
            style="color: black;"
          >
              Feedback 1 
          </div>
        </div>
      </div>
      <div
        data-testid="cell"
      >
        <div
          class="flex flex-col items-center justify-center relative"
        >
          <div
            class="relative w-fit mt-[-1.00px] font-bold text-black text-2xl leading-10 mb-2 whitespace-nowrap"
          >
            B1
          </div>
          <textarea
            class="w-full h-22 text-xs resize-none flex pt-4 px-4 pb-16 items-center bg-[#ececec] border-solid border border-black relative z-[1]"
            maxlength="148"
            placeholder="Isi sebab.."
            rows="1"
          />
          <div
            class="feedback-text mt-4"
            style="color: black;"
          >
              Feedback 2 
          </div>
        </div>
      </div>
      <div
        data-testid="cell"
      >
        <div
          class="flex flex-col items-center justify-center relative"
        >
          <div
            class="relative w-fit mt-[-1.00px] font-bold text-black text-2xl leading-10 mb-2 whitespace-nowrap"
          >
            C1
          </div>
          <textarea
            class="w-full h-22 text-xs resize-none flex pt-4 px-4 pb-16 items-center bg-[#ececec] border-solid border border-black relative z-[1]"
            maxlength="148"
            placeholder="Isi sebab.."
            rows="1"
          />
          <div
            class="feedback-text mt-4"
            style="color: black;"
          >
              Feedback 3 
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
